File: app/src/main/java/omniex/nl/omniex/ui/app/categories/CategoriesFragment.java
package omniex.nl.omniex.ui.app.categories;

import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.StaggeredGridLayoutManager;

import org.androidannotations.annotations.AfterViews;
import org.androidannotations.annotations.EFragment;
import org.androidannotations.annotations.ViewById;

import java.util.List;

import javax.inject.Inject;

import nl.omniex.omniexshopping.R;
import nl.omniex.omniexshopping.data.model.products.Category;
import nl.omniex.omniexshopping.ui.adapters.CategoriesAdapter;
import nl.omniex.omniexshopping.ui.app.product.list.ProductsListFragment_;
import nl.omniex.omniexshopping.ui.base.BaseFragment;
import nl.omniex.omniexshopping.ui.base.BaseRecyclerAdapter;

@EFragment(R.layout.fragment_categories)
public class CategoriesFragment extends BaseFragment<CategoriesView, CategoriesPresenter> implements CategoriesView, BaseRecyclerAdapter.ItemClickListener<Category> {

    @ViewById(R.id.categories_rv)
    RecyclerView mCategoriesRv;

    @Inject
    CategoriesAdapter mCategoriesAdapter;

    @AfterViews
    void initCategoriesList(){
        mCategoriesAdapter.setItemClickListener(this);
        mCategoriesRv.setLayoutManager(new StaggeredGridLayoutManager(2, StaggeredGridLayoutManager.VERTICAL));
        mCategoriesRv.setAdapter(mCategoriesAdapter);
        getPresenter().getCategories();
    }

    @Override
    public void onCategoriesFetched(List<Category> categoryList) {
        mCategoriesAdapter.setItems(categoryList);
    }

    @Override
    public void onItemClick(Category category) {
        goToFragment(ProductsListFragment_.builder().mIsBestSellersList(false).mCategoryId(category.getCategoryId()).build(), true);
    }
}

File: app/src/main/java/omniex/nl/omniex/ui/app/categories/CategoriesFragment.kt
package omniex.nl.omniex.ui.app.categories

import android.support.v7.widget.RecyclerView
import android.support.v7.widget.StaggeredGridLayoutManager
import omniex.nl.omniex.R
import omniex.nl.omniex.data.model.products.Category
import omniex.nl.omniex.ui.adapters.CategoriesAdapter
import omniex.nl.omniex.ui.app.product.list.ProductsListFragment_
import omniex.nl.omniex.ui.base.BaseFragment
import omniex.nl.omniex.ui.base.BaseRecyclerAdapter
import org.androidannotations.annotations.AfterViews
import org.androidannotations.annotations.EFragment
import org.androidannotations.annotations.ViewById
import javax.inject.Inject

@EFragment(R.layout.fragment_categories)
open class CategoriesFragment : BaseFragment<CategoriesView, CategoriesPresenter>(), CategoriesView, BaseRecyclerAdapter.ItemClickListener<Category> {

    @ViewById(R.id.categories_rv)
    lateinit var mCategoriesRv: RecyclerView

    @Inject
    lateinit var mCategoriesAdapter: CategoriesAdapter

    @AfterViews
    internal fun initCategoriesList() {
        mCategoriesAdapter!!.setClickListener(this)
        mCategoriesRv!!.layoutManager = StaggeredGridLayoutManager(2, StaggeredGridLayoutManager.VERTICAL)
        mCategoriesRv!!.adapter = mCategoriesAdapter
        getPresenter().getCategories()
    }

    override fun onCategoriesFetched(categoryList: List<Category>) {
        mCategoriesAdapter!!.setItems(categoryList)
    }

    override fun onItemClick(category: Category?) {
        goToFragment(ProductsListFragment_.builder().mIsBestSellersList(false).mCategoryId(category!!.categoryId).build(), true)
    }
}
